File: Clients/TinyDB.php
<?php

	class TinyDB {
		private $endpoint = null;
		private $database = '';
		private $user = ''; 
		private $pass = '';
		
		function __construct($host, $port, $ssl=false) {
			$this->endpoint = 'http'.($ssl ? 's' : '').'://'.$host.':'.strval($port).'/';
		}
		
		function db($database) {
			$this->database = strval($database);
			return $this;
		}
		
		function auth($username, $password) {
			$this->user = strval($username);
			$this->pass = strval($password);
			return $this;
		}
		
		function query($text) {
			try {
				$ch = curl_init();
				curl_setopt($ch, CURLOPT_URL, $this->endpoint);
				curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
				curl_setopt($ch, CURLOPT_POST, 1);
				curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, 0);
				curl_setopt($ch, CURLOPT_SSL_VERIFYHOST, 0);
				curl_setopt($ch, CURLOPT_POSTFIELDS, $text); 
				curl_setopt($ch, CURLOPT_HTTPHEADER, [
					'Database: '.$this->database,
					'Authorization: '.base64_encode(implode("\0",[$this->user,$this->pass])),
				]);
				$data = json_decode(curl_exec($ch));
				
				if (array_key_exists('data',$data) && !array_key_exists('err',$data)) {
					return $data->data;
				} else {
					throw new Error($data->err);
				}
			} catch (Exception $e) {
			}
			
			return null;
		}
	}
	
?>

File: Clients/TinyDB.py
import json
import requests
import base64

class TinyDB:
    endpoint = ''
    database = ''
    username = ''
    password = ''
    
    def __init__(self, host, port, ssl=False):
        self.endpoint = 'http'+('s' if ssl else '')+'://'+host+':'+str(port)+'/'
    
    def db(self, database):
        self.database = database
        return self
        
    def auth(self, username, password):
        self.username = username
        self.password = password
        return self
        
    def query(self, text):
        headers = {
            'Content-Type': 'text/plain',
            'Database': self.database,
            'Authorization': base64.b64encode((self.username+'\0'+self.password).encode('utf-8')).decode('utf-8'),
        }
        
        res = requests.post(self.endpoint, text, headers=headers)
        obj = json.loads(res.text)
        
        if 'err' in obj:
            raise Exception(obj['err'])
            
        return obj['data'];